src/coreclr/jit/compiler.cpp

```diff
@@ -1985,19 +1985,7 @@ void Compiler::compDone()
 void* Compiler::compGetHelperFtn(CorInfoHelpFunc ftnNum,        /* IN  */
                                  void**          ppIndirection) /* OUT */
 {
-    void* addr;
-
-    if (info.compMatchedVM)
-    {
-        addr = info.compCompHnd->getHelperFtn(ftnNum, ppIndirection);
-    }
-    else
-    {
-        // If we don't have a matched VM, we won't get valid results when asking for a helper function.
```

Comment:
I do not think it is safe to remove this. Some helpers are architecture specific and the VM will assert when you ask for them on a different architecture.

Reply:
It helped me to diagnose an arm64 jit issue, I just forgot to remove it before the check in. Removed